src/mlpack/methods/ann/layer/batch_norm_impl.hpp

```diff
@@ -30,6 +30,22 @@ BatchNorm<InputDataType, OutputDataType>::BatchNorm() :
 {
   // Nothing to do here.
 }
+
+template<typename InputDataType, typename OutputDataType>
+BatchNorm<InputDataType, OutputDataType>::BatchNorm(
+  const BatchNorm& layer) :
+    size(layer.size),
+    eps(layer.eps),
+    gamma(layer.gamma),
+    beta(layer.beta),
+    weights(layer.weights),
+    count(layer.count),
+    runningMean(layer.runningMean),
+    runningVariance(layer.runningVariance)
+{
+  Reset();
```

Comment:
I am using `Reset()` In copy constructor every layer.

Reply:
I'm not sure this is necessary though---shouldn't `FFN`'s copy constructor be doing that after all the layers are copied?  I think in many cases calling `Reset()` will not work correctly for layers at this point too---for the `Linear` layer, when `Reset()` is called, the `weight` and `bias` matrices are made to be aliases of the `weights` matrix.  But `weights` will be set by `FFN`'s copy constructor, so calling `Reset()` on a `Linear` layer during its copy constructor is too early---we have to wait until `FFN` sets `weights` (and then we can just have `FFN`'s copy constructor call `Reset()` on each layer at that time).